File: backend/app/schemas/student.py
from pydantic import BaseModel, Field


class JobATSAnalysis(BaseModel):
    job_id: str
    job_title: str
    ats_score: float = Field(ge=0, le=100)
    keyword_match_percent: float = Field(ge=0, le=100)
    similarity_score: float = Field(ge=0, le=100)
    suitability: str
    missing_keywords: list[str]
    matched_keywords: list[str]
    job_skills: list[str] = Field(default_factory=list)


class SuggestedRole(BaseModel):
    role_title: str
    confidence: float
    reason: str
    matched_skills: list[str] = Field(default_factory=list)


class StudentResumeResponse(BaseModel):
    id: str
    filename: str
    name: str | None = None
    email: str | None = None
    skills: list[str]
    created_at: str


class StudentATSReportResponse(BaseModel):
    resume_id: str
    analysis: JobATSAnalysis


class StudentJobMatchesResponse(BaseModel):
    resume_id: str
    matches: list[JobATSAnalysis]


class SuggestedRolesResponse(BaseModel):
    resume_id: str
    roles: list[SuggestedRole]

File: frontend/src/api/student.ts
import { apiClient } from './client';

export interface StudentResume {
  id: string;
  filename: string;
  name: string | null;
  email: string | null;
  skills: string[];
  created_at: string;
}

export interface JobATSAnalysis {
  job_id: string;
  job_title: string;
  ats_score: number;
  keyword_match_percent: number;
  similarity_score: number;
  suitability: string;
  missing_keywords: string[];
  matched_keywords: string[];
  job_skills: string[];
}

export interface SuggestedRole {
  role_title: string;
  confidence: number;
  reason: string;
  matched_skills: string[];
}

export async function fetchMyResume() {
  const { data } = await apiClient.get<StudentResume | null>('/student/resume');
  return data;
}

export async function uploadMyResume(file: File) {
  const formData = new FormData();
  formData.append('file', file);
  const { data } = await apiClient.post('/student/resume/upload', formData, {
    headers: { 'Content-Type': 'multipart/form-data' },
  });
  return data;
}

export async function fetchStudentJobs() {
  const { data } = await apiClient.get<{ id: string; filename: string; skills: string[]; created_at: string }[]>(
    '/student/jobs',
  );
  return data;
}

export async function analyzeJob(jobId: string) {
  const { data } = await apiClient.get<{ resume_id: string; analysis: JobATSAnalysis }>('/student/analyze', {
    params: { job_id: jobId },
  });
  return data;
}

export async function fetchJobMatches() {
  const { data } = await apiClient.get<{ resume_id: string; matches: JobATSAnalysis[] }>('/student/job-matches');
  return data;
}

export async function fetchSuggestedRoles() {
  const { data } = await apiClient.get<{ resume_id: string; roles: SuggestedRole[] }>('/student/suggested-roles');
  return data;
}
